File: ProjectEuler14/ProjectEuler14.java
public class ProjectEuler14{
	public long evenOrOdd(long number){
		if (number == 1){
			return 0;
		}
		else if (number % 2 == 0){
			return number / 2;
		}
		else{
			return number * 3 + 1;
		}
	}
	public static void main(String[] args){
		long number = 1;
		ProjectEuler14 call = new ProjectEuler14();
		long neededNumber = 0;
		long checkingNumber;
		int maxCount = 0;
		int count = 0;
		while (number < 1000000){
			checkingNumber = number;
			while (call.evenOrOdd(checkingNumber) != 0){
				count += 1;
				checkingNumber = call.evenOrOdd(checkingNumber);
			}
			if (count > maxCount){
				maxCount = count;
				neededNumber = number;
			}
			number += 1;
			count = 0;
		}
		System.out.println(neededNumber);
	}
}

File: ProjectEuler14/ProjectEuler14.py
def checkEvenOdd(number):
	if number == 1:
		return 0
	elif number % 2 == 0:
		return number // 2
	else:
		return 3 * number + 1

def findSequence():
	count = 1
	number = 2
	needed_number = 0
	maxCount = 0
	while number < 1000000:
		checkingNumber = number
		while checkEvenOdd(checkingNumber) != 1:
			checkingNumber = checkEvenOdd(checkingNumber)
			count += 1
		if count > maxCount:
			needed_number = number
			maxCount = count
		count = 0
		checkingNumber = 0
		number += 1
	return needed_number
print(findSequence())